concordia/views.py

```diff
@@ -145,6 +145,19 @@ def form_valid(self, form):
         return super().form_valid(form)
 
 
+class HomeView(TemplateView):
+    template_name = "home.html"
+
+    def get_context_data(self, *kws):
+        queryset = Campaign.objects.published().order_by("title")[:4]
```

Comment:
Should we hard-code the `[:4]` in here or let the template do the slicing so we don't need to touch the view if it changes?

Reply:
I didn't know the template could do that. It would be preferable to put it all in the template.

Reply:
Hmmm, yes. Is this a requirement that we always break it into two even rows rather than letting the browser wrap it?

https://getbootstrap.com/docs/4.1/layout/grid/#equal-width-multi-row suggests that we can do this with a simple `{% if forloop.count == column_count %}<div class="w-100"></div>{% endif %}`  which would cut down on the template complexity but doesn't help with the need to calculate that value. We do have that `custom_math` template tag library so we could simply put in `campaigns|length|multiply_by:0.5`.

Reply:
I think it's fine if they're not evenly distributed - if the browser decides. Your implementation is much cleaner and this way we don't have to modify anything except template code if we decide to change the number of campaigns we want to list on the home page. I changed it back to 6 since the flex implementation works for anything up to 6 items. So we don't need to change anything in the code unless we decide we want to show more than 6.

Reply:
It's also super-easy to do consistent columns using CSS Grid:

```diff
diff --git a/concordia/static/css/base.css b/concordia/static/css/base.css
index 950e6fd..8e71402 100644
--- a/concordia/static/css/base.css
+++ b/concordia/static/css/base.css
@@ -1898,3 +1898,9 @@ div.nav-secondary {
 .image-icon.itunes {
     background-position: -114px 0;
 }
+
+.section-homepage #campaign-list {
+    display: grid;
+    grid-template-columns: repeat(2, 1fr);
+    grid-gap: 0.6rem;
+}
diff --git a/concordia/templates/home.html b/concordia/templates/home.html
index fdf5e06..dd7b331 100644
--- a/concordia/templates/home.html
+++ b/concordia/templates/home.html
@@ -47,12 +47,12 @@
             </div>
         </div>
     </div>
-    <div class="row my-5">
+    <div class="row my-5 justify-content-center">
         <div class="col-12 mb-half">
             <h2 class="px-default px-md-0 ml-md-default">Campaigns: <small>Choose what to transcribe</small></h2>
         </div>
 
-        <div class="d-flex flex-wrap col-12 justify-content-around">
+        <div id="campaign-list" class="mx-auto">
             {% for campaign in campaigns %}
                 <div class="text-center campaign m-1">
                     <a href="{% url 'transcriptions:campaign' campaign.slug %}">
```

Maybe what we should do is add the grid bit above so we get a nice 2 column layout on modern browsers and IE11 falls back to flexbox?

Reply:
I like having 3 columns and 2 rows when there are six items - this would make it always 2 columns? I don't think I want that, but it's really up to @elainekamlley and @colefemeno

Reply:
That would always set two columns. We could either make it responsive (i.e. set a threshold for going from 1 to 2 to 3 columns) or do the calculation based on the current number of campaigns.